README.md

```diff
@@ -11,8 +11,599 @@
+    - [Additional Query Helper Functions](#additional-query-helper-functions)
+  - [Type safe Patches](#type-safe-patches)
+  - [Automatic type narrowing](#automatic-type-narrowing)
+    - [Field Name Limitations](#field-name-limitations)
+  - [ForgeRock API Documentation](#forgerock-api-documentation)
+- [Getting Started](#getting-started)
+- [Configuration](#configuration)
+# Features
+
+## TypeScript Type Code Generation
+
+The foundational feature is the TypeScript type code generation, all the other features rely on the generated types to function correctly.
+
+The [gen-idm-ts-types.js](bin/gen-idm-ts-types.js) script parses the Managed Object's (`managed.json`) and connector files (`provisioner.openicf-*.json`), and then generates the Typescript types.
+
+
+A snippet of a simple `managed.json` file:
+```json
+{
+    "$schema" : "http://forgerock.org/json-schema#",
+    "type" : "object",
+    "title" : "User",
+    "icon" : "fa-user",
+    "properties" : {
+        "_id" : {
+            "description" : "User ID",
+            "type" : "string"
+        },
+        "userName" : {
+            "title" : "Username",
+            "description" : "Username",
+            "type" : "string"
+        },
+        "password" : {
+            "title" : "Password",
+            "description" : "Password",
+            "type" : "string"
+        },
+        "givenName" : {
+            "title" : "First Name",
+            "description" : "First Name",
+            "type" : "string",
+            "searchable" : true,
+            "userEditable" : true,
+            "usageDescription" : null,
+            "isPersonal" : true
+        }
+    }
+}
+```
+
+And a portion of the resulting Typescript type:
+
+```ts
+export type ManagedUserDefaults = {
+  _tag?: "managed/user";
+
+  /**
+   * User ID
+   */
+  _id?: string;
+
+  /**
+   * Username
+   */
+  userName: string;
+
+  /**
+   * Password
+   */
+  password?: string;
+
+  /**
+   * First Name
+   */
+  givenName: string;
+}
+```
+
+### Supported Scenarios
+
+Code generation supports the following scenarios:
+
+| Scenario | Notes |
+| -------- | ----- |
+| Basic types | `string`, `number`/`integer`, `boolean`, `object`, `array` |
+| [Complex Objects](#complex-objects) | Objects that have defined properties, are generated as separate sub-types, and have unlimited levels of nesting, more info [below](#complex-objects). |
+| [Relationships](#relationships) | Fields that are [relationships](#relationships) simply use the target relationship as the type wrapped in a `ReferenceType`. It gracefully degrades when it can't find the target type to the `Record` type, this usually happens when referencing `internal` types which are not present in `managed.json`. |
+| Nullable fields | Fields marked as _nullable_ in the schema have add `null` as a Typescript [union type](https://www.typescriptlang.org/docs/handbook/2/everyday-types.html#union-types), e.g. `string \| null`. |
```

Comment:
nit: "have add `null` as a" -> "are unioned with `null`" ?

Reply:
Thanks for picking that up